src/UrbeArenaGladiators.sol

```diff
@@ -144,40 +165,79 @@ contract UrbeArenaGladiators is ERC721Enumerable, Ownable {
             !isDeath[targetTokenId],
             "You can't attack a gladiator that is already dead!"
         );
-        numberOfAttacksReceived[targetTokenId] += 1;
-        numAttacksEverReceived[targetTokenId] += 1;
-        // if this gladiator is receiving more attacks than the current highests, we empty the highests and we append this one
-        // if this gladiator is receiving the same attacks than the current highests, we append it to the highests
-        if (
-            _highestNumAttacksReceived <= numberOfAttacksReceived[targetTokenId]
-        ) {
+        //If the target is the died dude of the previous game, instead of attacking him he definitely kills him.
+        if (targetTokenId == likelyToDieId && isFightClosable()) {
+            closeDailyFight();
+        } else {
+            numberOfAttacksReceived[targetTokenId] += 1;
+            numAttacksEverReceived[targetTokenId] += 1;
+            // if this gladiator is receiving more attacks than the current highests, we empty the highests and we append this one
+            // if this gladiator is receiving the same attacks than the current highests, we append it to the highests
             if (
-                _highestNumAttacksReceived <
+                _highestNumAttacksReceived <=
                 numberOfAttacksReceived[targetTokenId]
             ) {
-                _highestNumAttacksReceived = numberOfAttacksReceived[
-                    targetTokenId
-                ];
-                delete mostAttackedCurrently;
+                if (
```

Comment:
Tutto questo if o magari solo gli annidati li metterei in funzioni separati per aumentare la leggibilità del codice